internal/providers/tree_artifacts.bzl

```diff
@@ -29,8 +29,8 @@ DirectoryFilePathInfo = provider(
 )
 
 def _directory_file_path(ctx):
-    if not ctx.file.directory.is_directory:
-        fail("directory attribute must be created with Bazel declare_directory (TreeArtifact)")
+    if not ctx.file.directory.is_source and not ctx.file.directory.is_directory:
```

Comment:
but it could be a source file, huh? I guess we could actually run an action here that just does the equiv of `[[ -d "ctx.file.directory" ]] || false`

Reply:
yes. unfortunately I don't know of a way to differentiate between a source file and a source directory. that seems like it needs a fix upstream in bazel proper